Modules/posixmodule.c

```diff
@@ -6805,6 +6805,9 @@ posix_read(PyObject *self, PyObject *args)
         return posix_error();
     }
     Py_BEGIN_ALLOW_THREADS
+    if (size > _PY_READ_MAX) {
+        size = _PY_READ_MAX;
```

Comment:
Hum. It seems inefficient to allocate size bytes and only later pass _PY_READ_MAX. This truncation should be done *before* PyString_FromStringAndSize().

Reply:
ping @matrixise